### PR TITLE
Fix Not Allowing to Pass `null` as Variable Input Value

### PR DESCRIPTION
**Pull request by DimuthuMadushan**
## Purpose

Fixes: [#2696](https://github.com/ballerina-platform/ballerina-standard-library/issues/2696)

## Examples
N/A

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: ballerina/variable_validator.bal [hunk 1]
```diff
@@ -242,21 +241,23 @@ class VariableValidator {
...
-    public isolated function setArgumentValue(json value, parser:ArgumentNode argument,
-                                              parser:VariableDefinitionNode varDef) {
-        if argument.getKind() == parser:T_INPUT_OBJECT {
-            argument.setVariableValue(value);
-        } else if argument.getKind() == parser:T_LIST {
-            argument.setVariableValue(value);
-        } else if argument.getKind() == parser:T_IDENTIFIER {
+    isolated function setArgumentValue(json value, parser:ArgumentNode argument, string variableTypeName,
+                                       __Type variableType) {
+        if getOfType(variableType).name.toString() == UPLOAD {
```
Comment (ThisaruGuruge) [inline ballerina/variable_validator.bal:246]: Why do we need to use `toString()` here? Shall we check the usage of `toString()` and try to remove/reduce the usage of `toString()`?

File: ballerina/validator_visitor.bal [hunk 2]
```diff
@@ -345,6 +351,9 @@ class ValidatorVisitor {
                         self.updatePath(listItemInputValue.name);
                         self.validateListVariableValue(listItemValue, listItemInputValue, location, fieldName);
                         self.removePath();
+                    } else if listItemValue is () {
+                        string expectedTypeName = getOfType(listItemInputValue.'type).name.toString();
```
Comment (ThisaruGuruge) [inline ballerina/validator_visitor.bal:355]: Shall we remove the `toString()` here as well? We can use the `string?` as the type here.

Comment (DimuthuMadushan) [reply ballerina/validator_visitor.bal:355]: We may have to cast it then.

Comment (ThisaruGuruge) [reply ballerina/validator_visitor.bal:355]: Yes, we can use casting or typeguard, whatever suits here. Using `toString()` is not an ideal way to handle these cases. There are many complexities in using `toString()`.

File: ballerina/common_utils.bal [hunk 3]
```diff
@@ -166,6 +166,15 @@ isolated function getOfType(__Type schemaType) returns __Type {
...
 }
 
+isolated function getOfTypeName(__Type schemaType) returns string {
+    __Type? ofType = schemaType?.ofType;
+    if ofType is () {
+        return <string>schemaType.name;
+    } else {
+        return <string>getOfType(ofType).name;
+    }
+}
```
Comment (ThisaruGuruge) [inline ballerina/common_utils.bal:176]: Can't this be simplified like this?

```suggestion
isolated function getOfTypeName(__Type schemaType) returns string {
    return <string>getOfType(schemaType).name
}
```

We are using `getOfType()` function but here we are trying to do the first level of unwrapping inside this function, which is redundant.

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**